plugins/module_utils/foreman_helper.py

```diff
@@ -313,14 +313,26 @@ def __init__(self, **kwargs):
         mutually_exclusive = kwargs.pop('mutually_exclusive', []) + [['medium', 'kickstart_repository']]
         super(HostMixin, self).__init__(foreman_spec=foreman_spec, required_plugins=required_plugins, mutually_exclusive=mutually_exclusive, **kwargs)
 
-        if 'activation_keys' in self.foreman_params:
-            if 'parameters' not in self.foreman_params:
-                self.foreman_params['parameters'] = []
-            ak_param = {'name': 'kt_activation_keys', 'parameter_type': 'string', 'value': self.foreman_params.pop('activation_keys')}
-            self.foreman_params['parameters'].append(ak_param)
+    def run(self, **kwargs):
+        entity = self.lookup_entity('entity')
+
+        if not self.desired_absent:
+            if 'activation_keys' in self.foreman_params:
+                if 'parameters' not in self.foreman_params:
+                    parameters = [param for param in (entity or {}).get('parameters', []) if param['name'] != 'kt_activation_keys']
+                else:
+                    parameters = self.foreman_params['parameters']
```

Comment:
We should make sure "kt_activation_keys" is not in this, too.
Maybe we just say, it's a wron use of the module to have it in both places.

Reply:
Validate Params will do that for us

Reply:
Because we add a second instance?
:facepalm:

Reply:
Correct.